lightwood/data/timeseries_transform.py

```diff
@@ -136,30 +137,29 @@ def transform_timeseries(
                 make_preds = [True for _ in range(len(df_arr[i]))]
             df_arr[i]['__make_predictions'] = make_preds
 
-    if len(original_df) > 500:
+    if len(df_arr) > 4 and len(original_df) > 5000:
         # @TODO: restore possibility to override this with args
-        nr_procs = get_nr_procs(original_df)
+        nr_procs = min(get_nr_procs(original_df), len(original_df))
         log.info(f'Using {nr_procs} processes to reshape.')
-        pool = mp.Pool(processes=nr_procs)
-        # Make type `object` so that dataframe cells can be python lists
-        df_arr = pool.map(partial(_ts_to_obj, historical_columns=[oby] + tss.historical_columns), df_arr)
```

Comment:
If we're OK without using `_ts_to_obj`, then let's remove its source code, too 👍

Reply:
Nevermind, missed the usage outside of the conditional.